decidim-proposals/lib/decidim/content_renderers/proposal_renderer.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ContentRenderers
+    # A renderer that searches Global IDs representing proposals in content
+    # and replaces it with a link to their show page.
+    #
+    # e.g. gid://<APP_NAME>/Decidim::Proposals::Proposal/1
+    #
+    # @see BaseRenderer Examples of how to use a content renderer
+    class ProposalRenderer < BaseRenderer
```

Comment:
It's fine for the moment, but this should probably eventually be `ResourceRenderer` - related to #2872 .